modules/tinymce/src/plugins/image/main/ts/core/ImageSelection.ts

```diff
@@ -33,8 +35,16 @@ const getSelectedImage = (editor: Editor): HTMLElement => {
 
 const splitTextBlock = (editor: Editor, figure: HTMLElement) => {
   const dom = editor.dom;
-
-  const textBlock = dom.getParent(figure.parentNode, (node: Node) => !!editor.schema.getTextBlockElements()[node.nodeName], editor.getBody());
+  const textBlockElements: SchemaMap = Obj.filter(
+    editor.schema.getTextBlockElements(),
+    (_, parentElm) => !editor.schema.isValidChild(parentElm, "figure")
+  );
+
+  const textBlock = dom.getParent(
+    figure.parentNode,
+    (node: Node) => !!textBlockElements[node.nodeName],
+    editor.getBody()
+  );
```

Comment:
I think it would be good to have a test for this fix. Probably add one in `ImageSelectionTest.ts`?

- [ ] Add a test

Also, don't forget to update the [changelog](https://github.com/tinymce/tinymce/blob/develop/modules/tinymce/changelog.txt)